crates/uv-fs/src/lib.rs

```diff
@@ -19,13 +19,22 @@ mod path;
 pub async fn read_to_string(path: impl AsRef<Path>) -> std::io::Result<String> {
     use std::io::Read;
 
+    use encoding_rs_io::DecodeReaderBytes;
+
     let path = path.as_ref();
     if path == Path::new("-") {
         let mut buf = String::with_capacity(1024);
-        std::io::stdin().read_to_string(&mut buf)?;
+        DecodeReaderBytes::new(std::io::stdin()).read_to_string(&mut buf)?;
         Ok(buf)
     } else {
-        fs_err::tokio::read_to_string(path).await
+        let path = path.to_path_buf();
+        tokio::task::spawn_blocking(move || {
+            let mut buf = String::with_capacity(1024);
```

Comment:
What's your reasoning behind spawning a background task for reading the file. My understanding of the use case is that a) we read the file as input before there's much async work anyway, b) the files are small text files. That makes me wonder if it's worth to decode and read the file on another thread (scheduling overhead etc).

Reply:
It's an `async` function and file I/O is blocking. So IMO, that means it should be spawned as a blocking task. Moreover, the code being replaced was also spawning a blocking task (it was just done inside of `tokio`), so this is maintaining that status quo. It's plausible that this isn't necessary in a holistic sense, but this particular routine isn't necessarily limited to use during startup.

Reply:
In general, i agree, in this specific place we're not doing anything else in parallel (we're not a webserver where we have other clients meanwhile) and the work is tiny, so i'd go for simplicity. I'd probably do a `fs_err::tokio::read` and then convert it using the other reader, even the double read should be far to small to show up in profiles.

Reply:
I went with @konstin's suggestion.